src/assisted_test_infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -184,8 +184,20 @@ def _fill_tfvars(self, running=True):
         tfvars["bootstrap_in_place"] = self._config.bootstrap_in_place
 
         vips = self.get_ingress_and_api_vips()
-        tfvars["api_vip"] = self._config.api_vip or vips["api_vip"]
-        tfvars["ingress_vip"] = self._config.ingress_vip or vips["ingress_vip"]
+        if self._config.api_vips:
+            tfvars["api_vips"] = [i.get("ip") for i in self._config.api_vips]
+        else:
+            tfvars["api_vips"] = [i.get("ip") for i in vips["api_vips"]]
+
+        if self._config.ingress_vips:
+            tfvars["ingress_vips"] = [i.get("ip") for i in self._config.ingress_vips]
+        else:
+            tfvars["ingress_vips"] = [i.get("ip") for i in vips["ingress_vips"]]
```

Comment:
nit: should we encapsulate this logic in some helper methods/functions? i.e. `tfavars["api_vips"] = get_api_vips(vips)`

Reply:
Done as a single function for both